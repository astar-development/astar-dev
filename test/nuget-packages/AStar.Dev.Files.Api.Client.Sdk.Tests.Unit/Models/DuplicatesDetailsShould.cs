using AStar.Dev.Files.Api.Client.SDK.Models;
using AStar.Dev.Utilities;
using JetBrains.Annotations;

namespace AStar.Dev.Files.Api.Client.Sdk.Models;

[TestSubject(typeof(DuplicatesDetails))]
public class DuplicatesDetailsShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new DuplicatesDetails
           {
               DetailsLastUpdated = DateTime.MaxValue,
               FileSize           = 1234,
               Height             = 123,
               Width              = 456,
               DirectoryName      = "MockDirectory",
               FileAccessDetailId = 567,
               FileHandle         = "MockHandle",
               FileName           = "MockFileName",
               HardDeletePending  = true,
               Id                 = 1,
               Instances          = 2,
               IsImage            = true,
               LastViewed         = new DateTime(2025, 12, 23),
               MoveRequired       = true,
               SoftDeleted        = true,
               SoftDeletePending  = true
           }
           .ToJson()
           .ShouldMatchApproved();
}
