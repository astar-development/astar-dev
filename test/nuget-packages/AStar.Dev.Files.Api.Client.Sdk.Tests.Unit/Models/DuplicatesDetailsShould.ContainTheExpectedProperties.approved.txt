{
  "id": 1,
  "fileAccessDetailId": 567,
  "fileName": "MockFileName",
  "directoryName": "MockDirectory",
  "height": 123,
  "width": 456,
  "fileSize": 1234,
  "fileHandle": "MockHandle",
  "isImage": true,
  "instances": 2,
  "detailsLastUpdated": "9999-12-31T23:59:59.9999999",
  "lastViewed": "2025-12-23T00:00:00",
  "softDeleted": true,
  "softDeletePending": true,
  "moveRequired": true,
  "hardDeletePending": true,
  "fullNameWithPath": "MockDirectory/MockFileName"
}